### DISCUSSION TITLE
Trim rst/rth files

### DISCUSSION THREAD

**Discussion by michikyle** [pyansys/pymapdl-reader#91] [Ideas]
Hi, is it possible to add functionality to this python package that would allow for trimming of .rth/.rst files? Often my colleagues and I run ANSYS only to realize the results file is much too large to fit on our hard drives, and would love to trim the output file without needing to change the outres command and re-run the simulation.

**Comment by akaszynski** [comment]
That's a great idea. Right now, work for post processing is being shifted over to https://github.com/pyansys/pydpf-core, but if there's no way it can be added there we can consider adding it here.

For the time being, I'd recommend saving only the select results you need into an unstructured grid using [save_as_vtk](https://readerdocs.pyansys.com/user_guide/loading_results.html?highlight=vtk#ansys.mapdl.reader.rst.Result.save_as_vtk).

Pinging @rlagha for awareness.

**Reply by rlagha** [reply]
@michikyle in DPF, you can do filtering of results based on high/low pass operators, or extract results only on the skin of the mesh/named selections/element types and then saved to another file if needed (vtk, hdf5). and then you can reuse DPF operators to re-import these new result files into your tools.
please let us know if you need any help. 
@akaszynski

**Reply by michikyle** [reply]
Thanks for the quick reply! I'll look into pydpf, and see if that fits our needs. 

I guess as a small point of clarification, I mostly was interested in trimming out the misc results and contact results and leaving just stress/strain for a .rst or just temperature for a .rth. Also, a trim that performs the equivalent of outres,all,last instead of outres,all,all. But I'll let you know if I have more specific questions related to that. 

Regards,
Kyle

**Reply by rlagha** [reply]
This is interesting, you can split the misc data into homogeneous fields, same for ECT data, request them only at given time steps and stream them to their own result file. We can definitely help with examples.

**Comment by mikerife** [comment]
Hi @michikyle You may want to look into the RSPLIT or /AUX3 commands in MAPDL (PyMAPDL).  RSPLIT can be used in the /post1 processor to create a full set of results for a sub-set of elements (a component).  Then you can post process (in PyMAPDL/MAPDL) the sub-set result file as you wish.  The INRES command can be used to limit the specific results to the new rst file.

/AUX3 in MAPDL can be used to delete result sets in an existing result file.  Maybe /COPY the result file; delete the results sets you don't need in the copy; then bring back the now smaller rst copy for local post processing.

If you are interested see the MAPDL documentation and then create a new discussion on the PyMAPDL page if/when needed.

Mike

**Reply by FrancescoSausto** [reply]
Hello @mikerife, I'm currently trying PyMAPDL Reader to extract results from Ansys rst files. It works nicely for complete rst files, but it crashes when I'm trying to read databases obtained after a RSPLIT command in APDL. In particular, I received this error "ANSYS binary "61" not supported". Do you have any suggestion? 

Thank you so much for the support.

Francesco
